app/main.py

```diff
@@ -0,0 +1,187 @@
+import json
```

Comment:
I don't like your mess with the alternation of functions and global expressions. Let's clean this:
- Put the logic with app construction under `main` function
- Call `main` from `if __name__ == "__main__"`

Reply:
Right, added ```main``` . Also i decompose it into ```main_widgets```, ```layout``` and ```session```. But taking into attention your 'inline' correction below.. I hope this satisfies you or we will find a compromise =)